### ISSUE TITLE
"AUCMLoss" Tensorflow 

### ISSUE THREAD

**Issue by MailSuesarn** [Optimization-AI/LibAUC#29]
Is it possible that I will use AUCMLoss in the version of Tensorflow 2? Can you suggest to me how to implement AUCMLoss  with Tensorflow 2

**Comment by yzhuoning** [comment]
It is definitely possible, however, implementations may be complicated in the case of TensorFlow. First, you need to create three  new learnable variables, i.e., `a`, `b`, `alpha` in the loss function based on our `AUCMLoss` loss. Second, you need to modify these variables in TF optimizer during updating parameters, i.e., updating `a`, `b` by gradient descent and updating `alpha` by gradient ascent. Additionally, you need to maintain a list of variables (referred as reference variable **`v`** in the paper)  to compute accumulated average of model variables for regularization purpose. For more details, you can take a look at the [original paper](https://openaccess.thecvf.com/content/ICCV2021/papers/Yuan_Large-Scale_Robust_Deep_AUC_Maximization_A_New_Surrogate_Loss_and_ICCV_2021_paper.pdf).

We are planning on releasing some code in TensorFlow in the coming months. Feel free to star this repository to stay updated on the latest developments and news. Thanks!

**Comment by MailSuesarn** [comment]
> It is definitely possible, however, implementations may be complicated in the case of TensorFlow. First, you need to create three new learnable variables, i.e., `a`, `b`, `alpha` in the loss function based on our `AUCMLoss` loss. Second, you need to modify these variables in TF optimizer during updating parameters, i.e., updating `a`, `b` by gradient descent and updating `alpha` by gradient ascent. Additionally, you need to maintain a list of variables (referred as reference variable **`v`** in the paper) to compute accumulated average of model variables for regularization purpose. For more details, you can take a look at the [original paper](https://openaccess.thecvf.com/content/ICCV2021/papers/Yuan_Large-Scale_Robust_Deep_AUC_Maximization_A_New_Surrogate_Loss_and_ICCV_2021_paper.pdf).
> 
> We are planning on releasing some code in TensorFlow in the coming months. Feel free to star this repository to stay updated on the latest developments and news. Thanks!

**Comment by MailSuesarn** [comment]
thank you very much I will wait for your new version release. It's a great job.

**Comment by dkhonker** [comment]
I seem not to have found any content related to TensorFlow. May I ask if TensorFlow is supported now?
